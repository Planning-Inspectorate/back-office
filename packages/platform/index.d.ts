import './config/dotenv';
import './http/ttl';
import './util/CryptoUtils';
import './util/date';
import './util/filter';
import './util/url';
import './validators/date';
import './validators/postcode';

declare module '@pins/platform' {
	export * from './config/dotenv';
	export * from './http/ttl';
	export * from './util/CryptoUtils';
	export * from './util/date';
	export * from './util/filter';
	export * from './util/url';
	export * from './validators/date';
	export * from './validators/postcode';
}
