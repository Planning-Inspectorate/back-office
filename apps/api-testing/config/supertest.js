// @ts-nocheck
import supertest from 'supertest';
import config from './config.cjs';
import { Logger } from '../utils/logger.js';

const logger = new Logger();
logger.log('green', '************* Tests starting ****************');
logger.log('cyan', `Running tests against: ${config.baseUrl}`);
logger.log('green', '*********************************************');

const request = supertest(config.baseUrl);

export default request;
