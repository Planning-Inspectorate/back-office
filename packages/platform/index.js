export * from './http/hooks.js';
export * from './testing/assets/assets.js';
export * from './testing/fake.js';
export * from './testing/html-parser.js';
export * from './utils/date.js';
export * from './utils/dotenv.js';
export * from './utils/filter.js';
export * from './validators/date.js';
export * from './validators/postcode.js';
