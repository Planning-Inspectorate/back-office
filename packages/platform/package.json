{
  "name": "@pins/platform",
  "version": "0.0.0",
  "type": "module",
  "main": "index.js",
  "private": true,
  "engines": {
    "node": ">=16.14.0"
  },
  "scripts": {
    "clean": "rimraf .turbo node_modules dist",
    "lint:js": "npx eslint .",
    "lint:js:fix": "npx eslint . --fix",
    "tscheck": "npx tsc -p jsconfig.json --maxNodeModuleJsDepth 0"
  },
  "dependencies": {
    "date-fns": "2.28.0",
    "dotenv": "16.0.0",
    "got": "^12.5.2",
    "html": "^1.0.0",
    "lodash-es": "^4.17.21",
    "node-html-parser": "^5.3.3",
    "randexp": "^0.5.3"
  },
  "devDependencies": {
    "@types/html": "^1.0.1",
    "@types/lodash-es": "^4.17.6",
    "type-fest": "^2.12.2"
  }
}
