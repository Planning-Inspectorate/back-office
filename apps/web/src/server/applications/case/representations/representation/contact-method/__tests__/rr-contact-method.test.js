import { parseHtml } from '@pins/platform';
import nock from 'nock';
import supertest from 'supertest';
import { createTestEnvironment } from '../../../../../../../../testing/index.js';
import { representationFixture } from '../../../__fixtures__/representations.fixture.js';

const { app, installMockApi, teardown } = createTestEnvironment();
const request = supertest(app);

const mockCaseReference = { title: 'mock title', status: 'in test', reference: 'mock reference' };
const nocks = () => {
	nock('http://test/').get('/applications/1').reply(200, mockCaseReference);
	nock('http://test/').get('/applications/1/representations/1').reply(200, representationFixture);
};

describe('Representation contact method page', () => {
	beforeEach(installMockApi);
	afterEach(teardown);

	afterAll(() => {
		nock.cleanAll();
	});

	const baseUrl =
		'/applications-service/case/1/relevant-representations/contact-method?repId=1&repType=represented';

	describe('GET /applications-service/case/1/relevant-representations/contact-method', () => {
		beforeEach(async () => {
			nocks();
			await request.get('/applications-service/case-team');
		});

		it('should render the page', async () => {
			const response = await request.get(baseUrl);
			const element = parseHtml(response.text);

			expect(element.innerHTML).toMatchSnapshot();
		});

		it('should have required fields', async () => {
			const response = await request.get(baseUrl);
			const element = parseHtml(response.text);

			expect(element.innerHTML).toContain('Preferred contact method');
		});
	});

	describe('POST /applications-service/case/1/relevant-representations/contact-method', () => {
		beforeEach(async () => {
			nocks();
		});
		describe('Field validation:', () => {
			it('should show validation error if no option selected', async () => {
				const response = await request.post(baseUrl).send({});

				const element = parseHtml(response.text);

				expect(element.innerHTML).toContain('Select one option');
			});
		});
	});
});
