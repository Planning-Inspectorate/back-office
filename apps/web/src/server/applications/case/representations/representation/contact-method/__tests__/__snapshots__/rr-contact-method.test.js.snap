// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Representation contact method page GET /applications-service/case/1/relevant-representations/contact-method should render the page 1`] = `
"<main class=\\"govuk-main-wrapper \\" id=\\"main-content\\" role=\\"main\\">
    <p class=\\"govuk-body\\"><strong>Project name:</strong> mock title</p><span class=\\"govuk-caption-l\\">Add a representation</span>
    <div     class=\\"govuk-grid-row\\">
        <div class=\\"govuk-grid-column-one-half\\">
            <form method=\\"post\\" novalidate>
                <div class=\\"govuk-form-group\\">
                    <fieldset class=\\"govuk-fieldset\\">
                        <legend class=\\"govuk-fieldset__legend govuk-fieldset__legend--l\\">
                            <h1 class=\\"govuk-fieldset__heading\\"> Preferred contact method</h1>
                        </legend>
                        <div class=\\"govuk-radios\\" data-module=\\"govuk-radios\\">
                            <div class=\\"govuk-radios__item\\">
                                <input class=\\"govuk-radios__input\\" id=\\"contactMethod\\" name=\\"contactMethod\\"
                                type=\\"radio\\" value=\\"email\\">
                                <label class=\\"govuk-label govuk-radios__label\\" for=\\"contactMethod\\">Email</label>
                            </div>
                            <div class=\\"govuk-radios__item\\">
                                <input class=\\"govuk-radios__input\\" id=\\"contactMethod-2\\" name=\\"contactMethod\\"
                                type=\\"radio\\" value=\\"post\\">
                                <label class=\\"govuk-label govuk-radios__label\\" for=\\"contactMethod-2\\">Post</label>
                            </div>
                        </div>
                    </fieldset>
                </div>
                <button type=\\"submit\\" class=\\"govuk-button govuk-!-margin-top-3\\" data-module=\\"govuk-button\\">Save and continue</button>
            </form>
        </div>
        </div>
</main>"
`;
